### PR TITLE
Add support of function analyze for vendor scene

### PR DESCRIPTION
**Pull request by hnhbwlp**
Hi,

This PR attempts to  add support of the function offset analyze for vendor scene.

Even though the go mod model has been launched for a long time, there are still go programs compiled through the vendor model in many private scenarios. When analyzing the Entry and Exit offsets of a function, the name of the function needs to be split to support the above scenarios. For example:  

-   I want to hook function createHeaderFields, which is in "path/to/myproject/vendor/google.golang.org/grpc/internal/transport.(*http2Client).createHeaderFields"
-  If I use "google.golang.org/grpc/internal/transport.(*http2Client).createHeaderFields",  it will not find this function from elf prog.
   
image

- If I use "path/to/myproject/vendor/google.golang.org/grpc/internal/transport.(*http2Client).createHeaderFields". Even though it will find the function correctly, different paths need to be configured in different projects.

If add this PR, it will works correctly!
image

### CODE REVIEW

**Review by edeNFed — APPROVED** [review]
Great catch! 
Lgtm

File: pkg/process/analyze.go [hunk 1]
```diff
@@ -141,16 +142,23 @@ func (a *processAnalyzer) Analyze(pid int, relevantFuncs map[string]interface{})
 	}
 
 	for _, f := range symTab.Funcs {
+		fName := f.Name
+		// fetch short path of function for vendor scene
+		if strings.Contains(fName, "/vendor/") {
+			if paths := strings.Split(fName, "/vendor/"); len(paths) > 1 {
```
[the remark this answers is not in the document]
Comment (hnhbwlp) [reply pkg/process/analyze.go:148]: No, the result of splitting the path in the common vendor mode is shown in the figure below, and path[0] will no longer contain "vendor".

image
result:
image

File: pkg/process/analyze.go [hunk 2]
```diff
@@ -141,16 +142,23 @@ func (a *processAnalyzer) Analyze(pid int, relevantFuncs map[string]interface{})
 	}
 
 	for _, f := range symTab.Funcs {
+		fName := f.Name
+		// fetch short path of function for vendor scene
+		if strings.Contains(fName, "/vendor/") {
```
[the remark this answers is not in the document]
Comment (damemi) [reply pkg/process/analyze.go:147]: Sorry, I mean, do you need to check `if strings.Contains(fName, "/vendor/")` before you call `if paths := strings.Split(fName, "/vendor/"); len(paths) > 1`?

the `strings.Split` will return a single item list if the value isn't found, so `len(paths) > 1` will fail. My point is, calling `strings.Contains` before that is redundant.

[the remark this answers is not in the document]
Comment (hnhbwlp) [reply pkg/process/analyze.go:147]: Sorry, I didn't understand you very well.  It doesn't need 'contains' check.

File: pkg/process/analyze.go [hunk 3]
```diff
@@ -141,16 +142,23 @@ func (a *processAnalyzer) Analyze(pid int, relevantFuncs map[string]interface{})
...
+			if paths := strings.Split(fName, "/vendor/"); len(paths) > 1 {
+				fName = paths[1]
+			}
+		}
```
[the remark this answers is not in the document]
Comment (hnhbwlp) [reply pkg/process/analyze.go:151]: It looks much cleaner now than before, I'll fix it, thanks！

File: pkg/process/analyze.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (damemi) [reply pkg/process/analyze.go]: No problem :) I should have been clearer in my first comment

**Review by damemi — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Comment by MrAlias** [comment]
This needs to sign its commits or wait for https://github.com/open-telemetry/community/issues/1380 to not require this as a branch protection.

**Comment by hnhbwlp** [comment]
> This needs to sign its commits or wait for [open-telemetry/community#1380](https://github.com/open-telemetry/community/issues/1380) to not require this as a branch protection.

Yea, I have signed the commit to fix this issue.

**Pull request merged**